### PR TITLE
Malware investigation and response

### PR DESCRIPTION
**Pull request by David-BMS**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/42383
fixes: https://github.com/demisto/etc/issues/44842
fixes: https://github.com/demisto/etc/issues/44753
fixes: https://github.com/demisto/etc/issues/45452

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/CommonPlaybooks/Playbooks/Search_For_Hash_In_Sandbox_-_Generic.yml [hunk 1]
```diff
@@ -0,0 +1,724 @@
...
+    isautoswitchedtoquietmode: false
+  "5":
+    id: "5"
+    taskid: 7bb954a8-b5f5-4051-8472-3c0d52b01561
+    type: regular
+    task:
+      id: 7bb954a8-b5f5-4051-8472-3c0d52b01561
+      version: -1
+      name: Search for hash in Wildfire
+      description: Retrieves results for a file hash using WildFire.
```
[the remark this answers is not in the document]
Comment (David-BMS) [reply Packs/CommonPlaybooks/Playbooks/Search_For_Hash_In_Sandbox_-_Generic.yml]: OK

File: Packs/CommonPlaybooks/Playbooks/Search_For_Hash_In_Sandbox_-_Generic.yml [hunk 2]
```diff
@@ -0,0 +1,724 @@
...
+    isautoswitchedtoquietmode: false
+  "23":
+    id: "23"
+    taskid: d346e9ac-c1df-4ad8-8009-dd0b0a84affc
+    type: regular
+    task:
+      id: d346e9ac-c1df-4ad8-8009-dd0b0a84affc
+      version: -1
+      name: Get Wildfire PDF report
+      description: Retrieves results for a file hash using WildFire.
```
[the remark this answers is not in the document]
Comment (David-BMS) [reply Packs/CommonPlaybooks/Playbooks/Search_For_Hash_In_Sandbox_-_Generic.yml]: confirmed

File: Packs/CrowdStrikeFalcon/Playbooks/playbook-CrowdStrike_Falcon_-_False_Positive_Incident_Handling.yml [hunk 3]
```diff
@@ -0,0 +1,831 @@
...
+  value: {}
+  required: false
+  description: The name of the tag to apply to allowed indicators.
+  playbookInputQuery:
+- key: ApplyAllowIOCGlobally
+  value:
+    simple: "True"
+  required: false
+  description: |-
+    Indicates if adding to allow list is globally
```
[the remark this answers is not in the document]
Comment (David-BMS) [reply Packs/CrowdStrikeFalcon/Playbooks/playbook-CrowdStrike_Falcon_-_False_Positive_Incident_Handling.yml]: Makes sense

File: Packs/CrowdStrikeFalcon/Playbooks/playbook-CrowdStrike_Falcon_Malware_-_Investigation_and_Response.yml [hunk 4]
```diff
@@ -0,0 +1,1937 @@
...
+      playbookName: Search For Hash In Sandbox - Generic
+      type: playbook
+      iscommand: false
+      brand: ''
+    nexttasks:
+      '#none#':
+      - '83'
+    scriptarguments:
+      FileSha256:
+        simple: ${incident.filesha256}
```
Comment (ssokolovich) [inline Packs/CrowdStrikeFalcon/Playbooks/playbook-CrowdStrike_Falcon_Malware_-_Investigation_and_Response.yml:556]: should we add there a unique transformer?

File: Packs/CrowdStrikeFalcon/Playbooks/playbook-CrowdStrike_Falcon_SIEM_ingestion_-_Get_Incident_Data.yml [hunk 5]
```diff
@@ -0,0 +1,893 @@
...
+  description: The detection ID.
+  type: string
+- contextPath: CrowdStrike.Detection.ProcessStartTime
+  description: The start time of the process that generated the detection.
+  type: string
+- contextPath: Endpoint
+  description: The details of the endpoint.
+  type: string
+tests:
+- Test - CrowdStrike Falcon
```
Comment (ssokolovich) [inline Packs/CrowdStrikeFalcon/Playbooks/playbook-CrowdStrike_Falcon_SIEM_ingestion_-_Get_Incident_Data.yml:893]: should be here a test?

**Review by ssokolovich — CHANGES_REQUESTED** [review]
There are minor changes from my review - please check my comments.

File: Packs/CommonPlaybooks/Playbooks/Search_For_Hash_In_Sandbox_-_Generic.yml [hunk 6]
```diff
@@ -0,0 +1,722 @@
+id: Search For Hash In Sandbox - Generic
+version: -1
+fromversion: 6.5.0
```
[the remark this answers is not in the document]
Comment (David-BMS) [reply Packs/CommonPlaybooks/Playbooks/Search_For_Hash_In_Sandbox_-_Generic.yml:3]: Everything we release now is 6.5

File: Packs/CommonPlaybooks/Playbooks/playbook-Detonate_and_Analyze_File_-_Generic.yml [hunk 7]
```diff
@@ -0,0 +1,441 @@
...
+- contextPath: csfalconx.resource.sandbox
+  description: The Falcon X findings.
+  type: string
+- contextPath: csfalconx.resource.intel
+  description: The Falcon X intelligence results.
+  type: string
+- contextPath: WildFire.Report
+  description: The Wildfire findings.
+  type: string
+tests:
```
[the remark this answers is not in the document]
Comment (David-BMS) [reply Packs/CommonPlaybooks/Playbooks/playbook-Detonate_and_Analyze_File_-_Generic.yml:418]: Done

File: Packs/CommonPlaybooks/Playbooks/playbook-Detonate_and_Analyze_File_-_Generic.yml [hunk 8]
```diff
@@ -0,0 +1,441 @@
...
+      type: playbook
+      iscommand: false
+      brand: ""
+    nexttasks:
+      '#none#':
+      - "9"
+    scriptarguments:
+      AlertOS:
+        complex:
+          root: incident
```
[the remark this answers is not in the document]
Comment (David-BMS) [reply Packs/CommonPlaybooks/Playbooks/playbook-Detonate_and_Analyze_File_-_Generic.yml:57]: We only need this for FalconX
So its not passed on from the generic playbook as an input but as a Falcon X input

File: Packs/CommonPlaybooks/Playbooks/playbook-Detonate_and_Analyze_File_-_Generic.yml [hunk 9]
```diff
@@ -0,0 +1,443 @@
...
+    isautoswitchedtoquietmode: false
+  "2":
+    id: "2"
+    taskid: a10b7555-36b7-444e-874c-20c10a692311
+    type: playbook
+    task:
+      id: a10b7555-36b7-444e-874c-20c10a692311
+      version: -1
+      name: Wildfire Detonate and Analyze File
+      playbookName: Wildfire Detonate and Analyze File
```
[the remark this answers is not in the document]
Comment (David-BMS) [reply Packs/CommonPlaybooks/Playbooks/playbook-Detonate_and_Analyze_File_-_Generic.yml:99]: Thats a sub playbook name and naming convention so best not to provide something else there

File: Packs/CommonPlaybooks/Playbooks/playbook-Detonate_and_Analyze_File_-_Generic.yml [hunk 10]
```diff
@@ -0,0 +1,443 @@
...
+    isoversize: false
+    isautoswitchedtoquietmode: false
+  "6":
+    id: "6"
+    taskid: e64d0590-3e6c-4b99-8ded-da01909d2195
+    type: condition
+    task:
+      id: e64d0590-3e6c-4b99-8ded-da01909d2195
+      version: -1
+      name: Is Wildfile enabled?
```
[the remark this answers is not in the document]
Comment (David-BMS) [reply Packs/CommonPlaybooks/Playbooks/playbook-Detonate_and_Analyze_File_-_Generic.yml:245]: Fixed

File: Packs/CommonPlaybooks/Playbooks/playbook-Detonate_and_Analyze_File_-_Generic.yml [hunk 11]
```diff
@@ -0,0 +1,441 @@
...
```
[the remark this answers is not in the document]
Comment (David-BMS) [reply Packs/CommonPlaybooks/Playbooks/playbook-Detonate_and_Analyze_File_-_Generic.yml:245]: Fixed

File: Packs/CommonPlaybooks/Playbooks/playbook-Detonate_and_Analyze_File_-_Generic.yml [hunk 12]
```diff
@@ -0,0 +1,441 @@
...
+      playbookName: Mitre Attack - Extract Technique Information From ID
+      type: playbook
+      iscommand: false
+      brand: ""
+    nexttasks:
+      '#none#':
+      - "3"
+    scriptarguments:
+      TechniqueID:
+        simple: ${csfalconx.resource.sandbox.mitre_attacks.attack_id}
```
[the remark this answers is not in the document]
Comment (David-BMS) [reply Packs/CommonPlaybooks/Playbooks/playbook-Detonate_and_Analyze_File_-_Generic.yml:346]: Done

File: Packs/CommonPlaybooks/Playbooks/playbook-Detonate_and_Analyze_File_-_Generic.yml [hunk 13]
```diff
@@ -0,0 +1,441 @@
...
+    separatecontext: true
+    loop:
```
[the remark this answers is not in the document]
Comment (David-BMS) [reply Packs/CommonPlaybooks/Playbooks/playbook-Detonate_and_Analyze_File_-_Generic.yml:348]: Its not looping
image

File: Packs/Palo_Alto_Networks_WildFire/Playbooks/playbook-Wildfire_Detonate_and_Analyze_File.yml [hunk 14]
```diff
@@ -0,0 +1,368 @@
...
+  value: {}
+  required: false
+  description: The details of the file to detonate.
+  playbookInputQuery:
+outputs:
+- contextPath: WildFire.Report
+  description: The Wildfire findings.
+  type: string
+tests:
+- Detonate URL - WildFire v2.1 - Test
```
[the remark this answers is not in the document]
Comment (David-BMS) [reply Packs/Palo_Alto_Networks_WildFire/Playbooks/playbook-Wildfire_Detonate_and_Analyze_File.yml:364]: Done

File: Packs/CrowdStrikeFalcon/Playbooks/playbook-CrowdStrike_Falcon_Malware_-_Incident_Enrichment.yml [hunk 15]
```diff
@@ -0,0 +1,666 @@
...
+      incident:
+      - '6'
+    separatecontext: false
+    conditions:
+    - label: incident
+      condition:
+      - - operator: startWith
+          left:
+            value:
+              simple: incident.externalsystemid
```
[the remark this answers is not in the document]
Comment (David-BMS) [reply Packs/CrowdStrikeFalcon/Playbooks/playbook-CrowdStrike_Falcon_Malware_-_Incident_Enrichment.yml:143]: image

File: Packs/CrowdStrikeFalcon/Playbooks/playbook-CrowdStrike_Falcon_Malware_-_Incident_Enrichment.yml [hunk 16]
```diff
@@ -0,0 +1,666 @@
...
+          right:
+            value:
+              simple: inc
+          ignorecase: true
+    - label: detection
+      condition:
+      - - operator: startWith
+          left:
+            value:
+              simple: incident.externalsystemid
```
[the remark this answers is not in the document]
Comment (David-BMS) [reply Packs/CrowdStrikeFalcon/Playbooks/playbook-CrowdStrike_Falcon_Malware_-_Incident_Enrichment.yml:154]: image

File: Packs/CrowdStrikeFalcon/Playbooks/playbook-CrowdStrike_Falcon_Malware_-_Investigation_and_Response.yml (hunk 4 above)
Comment (David-BMS) [reply Packs/CrowdStrikeFalcon/Playbooks/playbook-CrowdStrike_Falcon_Malware_-_Investigation_and_Response.yml:556]: Done

File: Packs/CrowdStrikeFalcon/Playbooks/playbook-CrowdStrike_Falcon_SIEM_ingestion_-_Get_Incident_Data.yml (hunk 5 above)
Comment (David-BMS) [reply Packs/CrowdStrikeFalcon/Playbooks/playbook-CrowdStrike_Falcon_SIEM_ingestion_-_Get_Incident_Data.yml:893]: Done

File: Packs/CrowdStrikeFalconX/Playbooks/playbook-FalconX_Detonate_and_Analyze_File.yml [hunk 17]
```diff
@@ -0,0 +1,391 @@
...
+- key: File
+  value: {}
+  required: false
+  description: The details of the file to detonate.
+  playbookInputQuery:
+- key: AlertOS
+  value:
+    complex:
+      root: ${incident
+      accessor: sourceosversion}
```
[the remark this answers is not in the document]
Comment (David-BMS) [reply Packs/CrowdStrikeFalconX/Playbooks/playbook-FalconX_Detonate_and_Analyze_File.yml:358]: Fixed

**Comment by julieschwartz18** [comment]
Doc review complete for the playbook-Malware_SIEM_Ingestion_-_Get_Incident_Data_README.md file, updated descriptions. @David-BMS - wasn't the **EDRProductToUse** input name updated to **SIEMEDRProductToUse**?

**Comment by julieschwartz18** [comment]
Doc review complete

**Review by David-BMS — COMMENTED** [review]
@julieschwartz18 Please make sure to add the pack readme as @bmillerPaloAlto requested

File: Packs/MalwareInvestigationAndResponse/README.md [hunk 18]
```diff
@@ -0,0 +1,38 @@
+Automate the investigation of your endpoint malware alerts and incidents.
```
Comment (David-BMS) [inline Packs/MalwareInvestigationAndResponse/README.md:1]: @julieschwartz18 @bmillerPaloAlto Please review

**Review by ssokolovich — APPROVED** [review]

**Pull request merged**